### PR TITLE
ENH: Extend _repr_html_ method for Raw

### PR DESCRIPTION
**Pull request by vagechirkov**
Reference issue #6846 and PR #9161

Add more information in the `_repr_html_` output.

image

### CODE REVIEW

**Comment by agramfort** [comment]
I would just show the basenames of the filenames and represent the
measurement time range in hour / min / seconds

>

**Comment by vagechirkov** [comment]
> I would just show the basenames of the filenames and represent the measurement time range in hour / min / seconds
> […](#)

Here is an example:

image

File: mne/io/base.py [hunk 1]
```diff
@@ -1651,9 +1651,17 @@ def __repr__(self):  # noqa: D105
     def _repr_html_(self, caption=None):
         if not isinstance(caption, str):
             caption = 'Raw'
+        basenames = [os.path.basename(f) for f in self._filenames]
+        m, s = divmod(self.first_time, 60)
+        h, m = divmod(m, 60)
+        tmin = '{:d}:{:02d}:{:02d}'.format(int(h), int(m), int(s))
```
Comment (hoechenberger) [inline mne/io/base.py:1657]: Could you turn this into an f-string please?

File: mne/io/base.py [hunk 2]
```diff
@@ -1651,9 +1651,17 @@ def __repr__(self):  # noqa: D105
...
+        m, s = divmod(self._last_time, 60)
+        h, m = divmod(m, 60)
+        tmax = '{:d}:{:02d}:{:02d}'.format(int(h), int(m), int(s))
```
Comment (hoechenberger) [inline mne/io/base.py:1660]: Same: could you turn this into an f-string please?

**Comment by hoechenberger** [comment]
Can we get rid of the "Raw" title? I believe it's unnecessary.

Screen Shot 2021-03-20 at 13 09 12

**Comment by hoechenberger** [comment]
"Measurement date" should ideally include the time too :)

File: mne/data/html_templates.py [hunk 3]
```diff
@@ -59,9 +59,13 @@
...
 raw_template = Template("""
 {{info_repr[:-9]}}
+    <tr>
+        <th>Filenames</th>
+        <td>{{' , '.join(filenames)}}</td>
+    </tr>
     <tr>
         <th>Measurement time range</th>
-        <td>{{u'%0.2f' % tmin}} to {{u'%0.2f' % tmax}} sec.</td>
+        <td>{{tmin}} to {{tmax}} (hours:minutes:seconds)</td>
```
Comment (hoechenberger) [inline mne/data/html_templates.py:68]: let's make this more consise:
```suggestion
        <td>{{tmin}} to {{tmax}} (HH:MM:SS</td>
```

File: mne/data/html_templates.py [hunk 4]
```diff
@@ -59,9 +59,13 @@
 
 raw_template = Template("""
 {{info_repr[:-9]}}
+    <tr>
+        <th>Filenames</th>
+        <td>{{' , '.join(filenames)}}</td>
```
Comment (hoechenberger) [inline mne/data/html_templates.py:64]:
```suggestion
        <td>{{', '.join(filenames)}}</td>
```

**Comment by vagechirkov** [comment]
> "Measurement date" should ideally include the time too :)

image

**Comment by hoechenberger** [comment]
Much better!

I haven't looked at the generated HTML code, but I have the impression that there's either another line 
break or empty row now before the first one actually containing data:

Screen Shot 2021-03-20 at 13 34 58


Do you think this could be improved easily?

File: mne/data/html_templates.py [hunk 5]
```diff
@@ -59,9 +59,13 @@
...
 raw_template = Template("""
 {{info_repr[:-9]}}
+    <tr>
+        <th>Filenames</th>
+        <td>{{', '.join(filenames)}}</td>
+    </tr>
     <tr>
         <th>Measurement time range</th>
-        <td>{{u'%0.2f' % tmin}} to {{u'%0.2f' % tmax}} sec.</td>
+        <td>{{tmin}} to {{tmax}} (HH:MM:SS)</td>
```
Comment (hoechenberger) [inline mne/data/html_templates.py:68]: Just tried locally and I think this looks better
```suggestion
        <td>{{tmin}} – {{tmax}} (H:MM:SS)</td>
```

File: mne/data/html_templates.py [hunk 6]
```diff
@@ -59,9 +59,13 @@
 
 raw_template = Template("""
 {{info_repr[:-9]}}
+    <tr>
+        <th>Filenames</th>
+        <td>{{', '.join(filenames)}}</td>
+    </tr>
     <tr>
         <th>Measurement time range</th>
```
Comment (hoechenberger) [inline mne/data/html_templates.py:67]: Do you think we could change this to simply "Measurement time"? Would be slightly more concise … @agramfort WDYT?

**Comment by vagechirkov** [comment]
> Much better!
> 
> I haven't looked at the generated HTML code, but I have the impression that there's either another line
> break or empty row now before the first one actually containing data:
> 
> Screen Shot 2021-03-20 at 13 34 58
> 
> Do you think this could be improved easily?

I think I've fixed this now :) 

image

**Comment by hoechenberger** [comment]
Cool, looking good!

**Comment by agramfort** [comment]
regarding measurement time range I would rather show the duration eg (6min 13s or 8h 2min 28s). For example a sleep recording can last 10h hours and it would be spanned over different days.

**Comment by hoechenberger** [comment]
> regarding measurement time range I would rather show the duration eg (6min 13s or 8h 2min 28s). For example a sleep recording can last 10h hours and it would be spanned over different days.

So you're proposing

```
Duration      12:34:56 (hh:mm:ss)
```
?

I like this.

Also wondering if we should maybe include `info['subject_info']['his_id']` as well, if present:
```
Participant      Mrs X
```

**Comment by agramfort** [comment]
Yes and yes 👍

**Comment by vagechirkov** [comment]
image

**Review by hoechenberger — APPROVED** [review]

**Review by agramfort — APPROVED** [review]
LGTM provided you manage to make CIs green

**Comment by agramfort** [comment]
Thx @vagechirkov

**Comment by hoechenberger** [comment]
This is a real usability improvement, thanks a bunch!!

**Comment by vagechirkov** [comment]
> This is a real usability improvement, thanks a bunch!!

Do you think it makes sense to create similar HTML-based representations for other MNE-Python objects?

**Comment by hoechenberger** [comment]
> > This is a real usability improvement, thanks a bunch!!
> 
> 
> 
> Do you think it makes sense to create similar HTML-based representations for other MNE-Python objects?


I was actually thinking about exactly this, yes. Most relevant for me personally would be Epochs and Evoked; but having such a nice representation for more objects eventually could make total sense!!

**Comment by vagechirkov** [comment]
> > > This is a real usability improvement, thanks a bunch!!
> > 
> > 
> > Do you think it makes sense to create similar HTML-based representations for other MNE-Python objects?
> 
> I was actually thinking about exactly this, yes. Most relevant for me personally would be Epochs and Evoked; but having such a nice representation for more objects eventually could make total sense!!

Alright, I can do this. I would create a new issue for several objects to keep progress :)

**Comment by agramfort** [comment]
+1 for evoked and epochs

**Pull request merged**